hazelcast/src/main/java/com/hazelcast/internal/partition/operation/ShutdownResponseOperation.java

```diff
@@ -45,7 +60,12 @@ public void run() {
             if (logger.isFinestEnabled()) {
                 logger.finest("Received shutdown response from " + caller);
             }
-            partitionService.onShutdownResponse();
+
+            if (nodeEngine.getLocalMember().getUuid().equals(uuid)) {
```

Comment:
It seems to me that this handling can go wrong when user tries to cancel the rolling upgrade from 5.1 to 5.2. Then, the shutdown process of 5.2 members may not be successful with this handling.

Reply:
Need to enhance my TODO programming skills, now it should be ok.